### PR TITLE
Add XRT API xclGetHostBO for MPSoC only

### PR DESCRIPTION
**Pull request by mamin506**
Add a MPSoC specific XRT API, xclGetHostBO. This API could map a buffer, which is allocated on host side, to PS application. 
Add zocl_cma_create() kernel function.

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

File: src/runtime_src/driver/zynq/drm/zocl/zocl_drv.c [hunk 1]
```diff
@@ -408,6 +439,17 @@ static int zocl_drm_platform_probe(struct platform_device *pdev)
 	zdev->res_start  = res->start;
 	zdev->res_len    = resource_size(res);
 
+	zdev->host_mem = 0;
```
Comment (sonals) [inline src/runtime_src/driver/zynq/drm/zocl/zocl_drv.c]: Should this be 0xffffffffffffffff

File: src/runtime_src/driver/zynq/include/zynq_ioctl.h [hunk 2]
```diff
@@ -33,13 +33,14 @@
...
 	DRM_ZOCL_USERPTR_BO,
+    DRM_ZOCL_GET_HOST_BO,
 	DRM_ZOCL_MAP_BO,
 	DRM_ZOCL_SYNC_BO,
 	DRM_ZOCL_INFO_BO,
 	DRM_ZOCL_PWRITE_BO,
 	DRM_ZOCL_PREAD_BO,
 	DRM_ZOCL_PCAP_DOWNLOAD,
-  DRM_ZOCL_EXECBUF,
+    DRM_ZOCL_EXECBUF,
```
Comment (sonals) [inline src/runtime_src/driver/zynq/include/zynq_ioctl.h]: Minor formatting fix.

File: src/runtime_src/driver/zynq/include/xclhal2_mpsoc.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * xclGetHostBO() - Get Host allocated BO
+ *
+ * @handle:        Device handle
+ * @paddr:         Physical address
+ * @size:          Size of the Buffer
+ * Return:         Buffer Object handler
+ *
+ * Get host allocated buffer by physical address.
```
Comment (sonals) [inline src/runtime_src/driver/zynq/include/xclhal2_mpsoc.h]: Let's add a comment that this is WIP and not to be used by clients at the moment.

**Comment by mamin506** [comment]
retest this please

**Pull request merged**